packages/ui/src/icon/Icon.test.tsx

```diff
@@ -0,0 +1,100 @@
+import { render } from "@testing-library/react";
+// repo
+import { TESTID_ICON } from "@repo/constants/src/testids";
+// workspace
+import Icon, { IconType } from "./Icon";
+
+// test constants
+export const getIconTestId = (type: IconType | string) =>
+  `${TESTID_ICON}-${type}`;
+
+describe("Icon", () => {
+  // close
+  it("renders close icon with title", () => {
+    const { getByTestId, getByTitle } = render(
+      <Icon type="close" id="closeIcon" title="close the menu" />,
+    );
+    const closeIcon = getByTestId(getIconTestId("close"));
+    expect(closeIcon).toBeInTheDocument();
+    expect(closeIcon).toHaveAttribute("aria-labelledby", "closeIcon");
+    expect(getByTitle("close the menu")).toBeInTheDocument();
+  });
+  it("renders close icon without title", () => {
+    const { getByTestId, queryByTitle } = render(<Icon type="close" />);
+    expect(getByTestId(getIconTestId("close"))).toBeInTheDocument();
+    expect(queryByTitle("close the menu")).not.toBeInTheDocument();
+  });
+  // menu
+  it("renders menu icon with title", () => {
+    const { getByTestId, getByTitle } = render(
+      <Icon type="menu" id="menuIcon" title="open the menu" />,
+    );
+    const menuIcon = getByTestId(getIconTestId("menu"));
+    expect(menuIcon).toBeInTheDocument();
+    expect(menuIcon).toHaveAttribute("aria-labelledby", "menuIcon");
+    expect(getByTitle("open the menu")).toBeInTheDocument();
+  });
+  it("renders menu icon without title", () => {
+    const { getByTestId, queryByTitle } = render(<Icon type="menu" />);
+    expect(getByTestId(getIconTestId("menu"))).toBeInTheDocument();
+    expect(queryByTitle("open the menu")).not.toBeInTheDocument();
+  });
+  // arrowOutward
+  it("renders arrowOutward icon with title", () => {
+    const { getByTestId, getByTitle } = render(
+      <Icon
+        type="arrowOutward"
+        id="arrowOutwardIcon"
+        title="arrow pointing outward"
+      />,
+    );
+    const arrowOutwardIcon = getByTestId(getIconTestId("arrowOutward"));
+    expect(arrowOutwardIcon).toBeInTheDocument();
+    expect(arrowOutwardIcon).toHaveAttribute(
+      "aria-labelledby",
+      "arrowOutwardIcon",
+    );
+    expect(getByTitle("arrow pointing outward")).toBeInTheDocument();
+  });
+  it("renders arrowOutward icon without title", () => {
+    const { getByTestId, queryByTitle } = render(<Icon type="arrowOutward" />);
+    expect(getByTestId(getIconTestId("arrowOutward"))).toBeInTheDocument();
+    expect(queryByTitle("arrow pointing outward")).not.toBeInTheDocument();
+  });
+  // openInNew
+  it("renders openInNew icon with title", () => {
+    const { getByTestId, getByTitle } = render(
+      <Icon type="openInNew" id="openInNewIcon" title="open in a new tab" />,
+    );
+    const openInNewIcon = getByTestId(getIconTestId("openInNew"));
+    expect(openInNewIcon).toBeInTheDocument();
+    expect(openInNewIcon).toHaveAttribute("aria-labelledby", "openInNewIcon");
```

Comment:
Are we checking the value of `aria-labelledby` specifically in the tests that include a title to prove that it doesn't match the `title`?

Would it make sense to confirm the `aria-labelledby` value exists and is correct on on non-titled icon as well?

Reply:
As currently set up, the `aria-labelledby` attribute is only added on ones that also have a title so we're only checking for it in the ones where both attributes are provided. My current thought process is that I'm adding them together to add a11y text for screen readers. However, in some cases the icon could be decorative and not need to be read out (I'm thinking buttons with text AND and icon). So in those cases I wouldn't want the icon to have an aria-labelledby attribute. Thoughts?

Reply:
Makes sense!

Valid, yes. I think the only time we have to be sure that there is an `aria-labelledby` is in this title situation, when an icon adds extra information, or when it's an icon button. Solid.